carousel/core/models.py

```diff
@@ -218,7 +218,16 @@ def _initialize(self):
             if src_value:
                 value = src_model[layer] = src_value
             else:
-                value = dict(value['sources'])
+                srcmod, srcpkg = value.get('module'), value.get('package')
+                try:
+                    value = dict(value['sources'])
+                except ValueError:
+                    value = dict.fromkeys(value['sources'], {})
+                for src in value.viewkeys():
+                    if srcmod is not None:
+                        value[src]['module'] = srcmod
+                    if srcpkg is not None:
+                        value[src]['package'] = srcpkg
             # set layer attribute with model data
             setattr(self, layer, layer_cls(value))
```

Comment:
I get really confused about what ``value`` ends up being... and why we can always use it to instantiate ``layer_cls``.
This whole method is still quite abstract to me; but I'm making progress!